### PR TITLE
Support configuring init containers for components in TiDB cluster

### LINKED ISSUE #3695: Support configuring init containers for components in TiDB cluster
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

Support configuring init containers for components in TiDB cluster:
PD, TiDB, TiKV, TiFlash, TiCDC, Pump

Add field similar with https://github.com/pingcap/tidb-operator/blob/master/pkg/apis/pingcap/v1alpha1/types.go#L813:
```
// Init containers of the component.
	// +optional
	InitContainers []corev1.Container `json:"initContainers,omitempty"`
```
**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?

Closes #3695 
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

Add the spec below to TiDB/TiKV/TiFlash/PD spec:

```yaml
annotations:
  tidb.pingcap.com/sysctl-init: "true"
podSecurityContext:
  sysctls:
  - name: net.ipv4.tcp_keepalive_time
    value: "300"
  - name: net.ipv4.tcp_keepalive_intvl
    value: "75"
  - name: net.core.somaxconn
    value: "32768"
initContainers:
  - command:
      - /bin/sh
    image: busybox
    imagePullPolicy: IfNotPresent
    name: inittest
```

We can see TiKV init with 2 pods, shows the inject of init containers succeed.

```output
cluster1-discovery-7d8dc88cbb-mdc6l        1/1     Running    0          19s
cluster1-pd-0                              1/1     Running    0          19s
cluster1-pump-0                            1/1     Running    0          8s
cluster1-tikv-0                            0/1     Init:0/2   0          8s
tidb-controller-manager-69548c7756-kc49n   1/1     Running    0          63m
tidb-scheduler-75c5fb8fbf-b4f2h            2/2     Running    0          63m
```

The spec of TiKV Pods shows there are 2 initContainers as we designed:

```yaml
initContainers:
  - command:
    - sh
    - -c
    - sysctl -w net.ipv4.tcp_keepalive_time=300 net.ipv4.tcp_keepalive_intvl=75 net.core.somaxconn=32768
    image: busybox:1.26.2
    imagePullPolicy: IfNotPresent
    name: init
    resources: {}
    securityContext:
      privileged: true
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-qfdnz
      readOnly: true
  - command:
    - /bin/sh
    image: busybox
    imagePullPolicy: IfNotPresent
    name: inittest
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-qfdnz
      readOnly: true
```

After initilization, the status of Pods turn to Running.

```output
cluster1-discovery-7d8dc88cbb-mdc6l        1/1     Running   0          3m54s
cluster1-pd-0                              1/1     Running   0          3m54s
cluster1-pump-0                            1/1     Running   0          3m43s
cluster1-ticdc-0                           1/1     Running   0          3m6s
cluster1-tidb-0                            2/2     Running   0          3m6s
cluster1-tikv-0                            1/1     Running   0          3m43s
tidb-controller-manager-69548c7756-kc49n   1/1     Running   0          66m
tidb-scheduler-75c5fb8fbf-b4f2h            2/2     Running   0          66m
```

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support configuring init containers for components in TiDB cluster
```

### CODE REVIEW

File: pkg/manager/member/tidb_member_manager.go [hunk 1]
```diff
@@ -715,7 +715,7 @@ func getNewTiDBSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap)
 	podSpec.Containers = append(containers, baseTiDBSpec.AdditionalContainers()...)
 	podSpec.Volumes = append(vols, baseTiDBSpec.AdditionalVolumes()...)
 	podSpec.SecurityContext = podSecurityContext
-	podSpec.InitContainers = initContainers
+	podSpec.InitContainers = append(initContainers, baseTiDBSpec.InitContainers()...)
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidb_member_manager.go:718]: Please also support the other components, e.g. PD, TiKV, TiFlash, TiCDC, Pump.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@csuzhangxc PTAL.

**Review by dragonly — APPROVED** [review]
LGTM

**Pull request merged**